transform/webpack.common.js

```diff
@@ -0,0 +1,104 @@
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const ModuleFederationPlugin = require('webpack').container.ModuleFederationPlugin;
+const DefinePlugin = require('webpack').DefinePlugin;
+const NodePolyfillPlugin = require('node-polyfill-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const { dependencies } = require('./package.json');
+const path = require('path');
+
+const isPatternflyStyles = (stylesheet) =>
+  stylesheet.includes('@patternfly/react-styles/css/') ||
+  stylesheet.includes('@patternfly/react-core/');
+
+module.exports = () => {
+  return {
+    entry: path.resolve(__dirname, 'src', 'index.tsx'),
+    target: 'web',
+    devtool: 'source-map',
+    output: {
+      path: path.resolve(__dirname, 'dist'),
+      publicPath: 'auto',
+    },
+    resolve: {
+      extensions: ['.ts', '.tsx', '.js'],
+      fallback: {
+        http: require.resolve('stream-http'),
+        util: require.resolve('util'),
+        buffer: require.resolve('buffer'),
+      },
+    },
+    module: {
+      rules: [
+        {
+          test: /\.css$/,
+          include: isPatternflyStyles,
+          use: ['null-loader'],
+          sideEffects: true,
+        },
+        {
+          test: /\.css$/,
+          use: ['style-loader', 'css-loader'],
+        },
+        {
+          test: /\.tsx?$/,
+          loader: 'babel-loader',
+          exclude: /node_modules/,
+          options: {
+            presets: [
+              '@babel/preset-typescript',
+              ['@babel/preset-react', { runtime: 'automatic' }],
+            ],
+          },
+        },
+        {
+          test: /\.(ttf|eot|woff|woff2)$/,
+          type: 'asset/resource',
+        },
+      ],
+    },
+    plugins: [
+      new MiniCssExtractPlugin(),
+      new ModuleFederationPlugin({
+        name: 'transformStep',
+        filename: 'remoteEntry.js',
+        exposes: {
+          './TransformStep': './src/Components/TransformStep',
+        },
+        shared: {
+          ...dependencies,
+          react: {
+            singleton: true,
+            requiredVersion: dependencies['react'],
+          },
+          'react-dom': {
+            singleton: true,
+            requiredVersion: dependencies['react-dom'],
+          },
+          'react-i18next': {
```

Comment:
Unrelated to your PR I think, but I wonder if we are doing this for the step extension template, because we should really require anyone to be doing anything with i18n unless they want to.

Reply:
Filed an issue - https://github.com/KaotoIO/step-extension/issues/11